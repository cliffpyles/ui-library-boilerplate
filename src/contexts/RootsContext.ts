import React from "react";

// When toggling aria-hidden values, we only want to affect elements
// in the DOM that come from ComponentLibrary, so we track those elements in this
// context value. See FocusedContainer.js
export const RootsContext = React.createContext([]);
