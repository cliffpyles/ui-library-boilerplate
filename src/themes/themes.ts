// src/themes/themes.ts
import { buildTheme } from "../utils/buildTheme/buildTheme";

export const lightTheme = {
  background: 'white',
  color: 'black',
};

export const darkTheme = {
  background: 'black',
  color: 'white',
};

export const brandedTheme = buildTheme({
  global: {
    colors: {
      gunmetal: '#31393cff',
      blueCrayola: '#2176ffff',
      celestialBlue: '#33a1fdff',
      sunglow: '#fdca40ff',
      carrotOrange: '#f79824ff',
      'light-1': '#ffffff',
      'light-2': '#f7f7f7',
      'light-3': '#efefef',
      'light-4': '#e7e7e7',
      'light-5': '#dfdfdf',
      'light-6': '#d7d7d7',
      'light-7': '#cfcfcf',
      'light-8': '#c7c7c7',
      'light-9': '#bfbfbf',
      'light-10': '#b7b7b7',
      'dark-1': '#000000',
      'dark-2': '#080808',
      'dark-3': '#101010',
      'dark-4': '#181818',
      'dark-5': '#202020',
      'dark-6': '#282828',
      'dark-7': '#303030',
      'dark-8': '#383838',
      'dark-9': '#404040',
      'dark-10': '#484848'
    },
    font: {
      family: 'Arial, sans-serif',
      size: '16px',
      weight: '400',
    },
    lineHeight: '1.5',
    spacing: {
      none: '0',
      xxsmall: '4px',
      xsmall: '8px',
      small: '16px',
      medium: '24px',
      large: '32px',
      xlarge: '40px',
      xxlarge: '48px',
    },
    border: {
      color: '#000000',
      width: '1px',
      style: 'solid',
    },
    borderRadius: {
      none: '0',
      small: '4px',
      medium: '8px',
      large: '16px',
    },
    elevation: {
      none: 'none',
      xsmall: '0 1px 2px rgba(0, 0, 0, 0.05)',
      small: '0 2px 4px rgba(0, 0, 0, 0.1)',
      medium: '0 4px 8px rgba(0, 0, 0, 0.15)',
      large: '0 8px 16px rgba(0, 0, 0, 0.2)',
      xlarge: '0 16px 32px rgba(0, 0, 0, 0.25)'
    },
    focus: {
      outline: '1px solid blue',
    }
  },
  box: {
    background: '$global.colors.light-1',
    border: {
      color: '$global.border.color',
      width: '$global.border.width',
      style: '$global.border.style',
    },
    padding: {
      xxsmall: '$global.spacing.xxsmall',
      xsmall: '$global.spacing.xsmall',
      small: '$global.spacing.small',
      medium: '$global.spacing.medium',
      large: '$global.spacing.large',
      xlarge: '$global.spacing.xlarge',
      xxlarge: '$global.spacing.xxlarge',
    },
    margin: {
      xxsmall: '$global.spacing.xxsmall',
      xsmall: '$global.spacing.xsmall',
      small: '$global.spacing.small',
      medium: '$global.spacing.medium',
      large: '$global.spacing.large',
      xlarge: '$global.spacing.xlarge',
      xxlarge: '$global.spacing.xxlarge',
    },
    borderRadius: {
      none: '$global.borderRadius.none',
      small: '$global.borderRadius.small',
      medium: '$global.borderRadius.medium',
      large: '$global.borderRadius.large',
    },
    elevation: {
      none: '$global.elevation.none',
      xsmall: '$global.elevation.xsmall',
      small: '$global.elevation.small',
      medium: '$global.elevation.medium',
      large: '$global.elevation.large',
      xlarge: '$global.elevation.xlarge'
    }
  }
});

export const themes = {
  light: lightTheme,
  dark: darkTheme,
  branded: brandedTheme,
};
