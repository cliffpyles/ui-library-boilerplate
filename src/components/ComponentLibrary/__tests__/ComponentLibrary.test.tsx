import { useEffect } from "react";
import { render } from "@testing-library/react";
import { ComponentLibrary } from "..";
import { AnnounceContext } from "../../../contexts";
import { base } from "../../../themes/base";
import { MessageContext } from "../../../contexts/MessageContext";

const TestAnnouncer = ({ announce }) => {
  useEffect(() => announce("hello", "assertive"));
  return <div>hi</div>;
};

describe("ComponentLibrary", () => {
  test("basic", () => {
    const { container } = render(<ComponentLibrary />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("themeMode", () => {
    const { container } = render(<ComponentLibrary theme={base} themeMode="dark" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("cssVars", () => {
    const { container } = render(<ComponentLibrary cssVars>ComponentLibrary App</ComponentLibrary>);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("full", () => {
    const { container } = render(<ComponentLibrary full>ComponentLibrary App</ComponentLibrary>);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("full min", () => {
    const { container } = render(<ComponentLibrary full="min">ComponentLibrary App</ComponentLibrary>);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("background", () => {
    const { container } = render(
      <ComponentLibrary full background="#0000ff">
        ComponentLibrary App
      </ComponentLibrary>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test("announce", (done) => {
    const { container } = render(
      <ComponentLibrary>
        <AnnounceContext.Consumer>{(announce) => <TestAnnouncer announce={announce} />}</AnnounceContext.Consumer>
      </ComponentLibrary>
    );
    expect(container.firstChild).toMatchSnapshot();

    // no style, no need for expectPortal
    expect(document.body.querySelector("#grommet-announcer[aria-live]")).toMatchSnapshot();

    setTimeout(() => {
      // should clear the aria-live container
      expect(document.body.querySelector("#grommet-announcer[aria-live]")).toMatchSnapshot();
      done();
    }, 600); // wait the aria-live container to clear
  });

  test("messages", () => {
    const { container } = render(
      <ComponentLibrary
        messages={{
          messages: {
            test: {
              label: "My Label",
            },
          },
        }}
      >
        <MessageContext.Consumer>{({ format }) => format({ id: "test.label" })}</MessageContext.Consumer>
      </ComponentLibrary>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test("message format function", () => {
    const messages = {
      "test.label": "My Label",
    };
    const { container } = render(
      <ComponentLibrary full background="#0000ff">
        ComponentLibrary App
      </ComponentLibrary>,

      <ComponentLibrary messages={{ format: (opts) => messages[opts.id] }}>
        <MessageContext.Consumer>{({ format }) => format({ id: "test.label" })}</MessageContext.Consumer>
      </ComponentLibrary>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
