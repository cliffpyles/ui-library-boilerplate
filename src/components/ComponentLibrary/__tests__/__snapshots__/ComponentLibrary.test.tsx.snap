// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ComponentLibrary > announce 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  class="c0"
>
  <div>
    hi
  </div>
</div>
`;

exports[`ComponentLibrary > announce 2`] = `
<div
  aria-live="assertive"
  id="grommet-announcer"
  style="left: -100%; right: 100%; position: fixed; z-index: -1;"
>
  hello
</div>
`;

exports[`ComponentLibrary > background 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  background-color: #0000ff;
  color: #f8f8f8;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  width: 100vw;
  height: 100vh;
  overflow: auto;
}

<div
  class="c0"
>
  ComponentLibrary App
</div>
`;

exports[`ComponentLibrary > basic 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  class="c0"
/>
`;

exports[`ComponentLibrary > cssVars 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  --active: rgba(221, 221, 221, 0.5);
  --active-background: background-contrast;
  --active-text: text-strong;
  --black: #000000;
  --brand: #7D4CDB;
  --focus: #6FFFB0;
  --graph-0: accent-1;
  --graph-1: neutral-1;
  --graph-2: neutral-2;
  --graph-3: neutral-3;
  --graph-4: neutral-4;
  --placeholder: #AAAAAA;
  --selected: brand;
  --selected-background: brand;
  --selected-text: text-strong;
  --white: #FFFFFF;
  --accent-1: #6FFFB0;
  --accent-2: #FD6FFF;
  --accent-3: #81FCED;
  --accent-4: #FFCA58;
  --dark-1: #333333;
  --dark-2: #555555;
  --dark-3: #777777;
  --dark-4: #999999;
  --dark-5: #999999;
  --dark-6: #999999;
  --light-1: #F8F8F8;
  --light-2: #F2F2F2;
  --light-3: #EDEDED;
  --light-4: #DADADA;
  --light-5: #DADADA;
  --light-6: #DADADA;
  --neutral-1: #00873D;
  --neutral-2: #3D138D;
  --neutral-3: #00739D;
  --neutral-4: #A2423D;
  --status-critical: #FF4040;
  --status-error: #B30000;
  --status-warning: #FFAA15;
  --status-ok: #00C781;
  --status-unknown: #CCCCCC;
  --status-disabled: #CCCCCC;
}

<div
  class="c0"
>
  ComponentLibrary App
</div>
`;

exports[`ComponentLibrary > full 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  width: 100vw;
  height: 100vh;
  overflow: auto;
}

<div
  class="c0"
>
  ComponentLibrary App
</div>
`;

exports[`ComponentLibrary > full min 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  min-height: 100vh;
}

<div
  class="c0"
>
  ComponentLibrary App
</div>
`;

exports[`ComponentLibrary > message format function 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  background-color: #0000ff;
  color: #f8f8f8;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  width: 100vw;
  height: 100vh;
  overflow: auto;
}

<div
  class="c0"
>
  ComponentLibrary App
</div>
`;

exports[`ComponentLibrary > messages 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  class="c0"
>
  My Label
</div>
`;

exports[`ComponentLibrary > themeMode 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  class="c0"
/>
`;
